src/main/java/com/_up/megastore/data/utilities/PhoneNumberConversion.java

```diff
@@ -0,0 +1,27 @@
+package com._up.megastore.data.utilities;
+
+import com.google.i18n.phonenumbers.NumberParseException;
+import com.google.i18n.phonenumbers.PhoneNumberUtil;
+import com.google.i18n.phonenumbers.PhoneNumberUtil.PhoneNumberFormat;
+import com.google.i18n.phonenumbers.Phonenumber.PhoneNumber;
+
+public class PhoneNumberConversion {
+
+  /**
+   * Converts a phone number string to the E.164 format.
+   *
+   * @param phoneNumber the phone number string to be converted
+   * @return the phone number in E.164 format
+   * @throws RuntimeException if the phone number cannot be parsed
+   */
+  public static String convertPhoneNumber(String phoneNumber) {
+    PhoneNumberUtil phoneUtil = PhoneNumberUtil.getInstance();
+
+    try {
+      PhoneNumber numberProto = phoneUtil.parse(phoneNumber, "AR");
+      return phoneUtil.format(numberProto, PhoneNumberFormat.E164);
+    } catch (NumberParseException e) {
+      throw new RuntimeException("Failed to convert phone number: " + phoneNumber, e);
+    }
```

Comment:
Mantendría la misma excepción, no la cambiaría a un RuntimeException

Reply:
Entonces debería ignorar el error que obtengo de la excepción? 
Generalmente en estos casos, se loggea para saber internamente que fue lo que paso, o no?

Reply:
Al catchear la excepción lo que vos estas haciendo es que no lance un error al cliente retornando un error 500 y harás lo que vos necesités, supongamos que deseas loggear, en ese caso tendrias que hacer

```suggestion
    } catch (NumberParseException e) {
      logger.info("There was an error parsing the phone number")
    }
```

pero lo que estas haciendo en este codigo es catchear la excepcion del tipo NumberParseException para lanzar otra del tipo RuntimeException, para eso deja la misma que ya se lanzó previamente y podes en un ExceptionHandler manejar la exception de NumberParseException como vos quieras

Reply:
Es que yo dejaría que tire error en el cliente indicando que no se pudo parsear el numero. Aunque en este caso nunca pasaría porque lo estoy validando con una constraint.
Me podes tirar algún centro? El método parse ya me obliga a manejar la excepción (en la firma dice que tira la misma)

Reply:
¿Y una excepción personalizada? Creo que el problema de dejar la NumberParseException es que es [chequeada](https://openwebinars.net/blog/introduccion-a-poo-en-java-excepciones/#:~:text=de%20cada%20secci%C3%B3n.-,Tipos%20de%20excepciones%20en%20Java,en%20un%20bloque%20try%2Dcatch.) y ensucia todos los métodos desde el service hasta esta clase. Por otro lado, por más que en el cliente se valide la excepción que tiene que existir porque el back se podría probar con Postman o cualquier cliente distinto

Reply:
A ver, el flujo es el siguiente:
- Mandas el numero de teléfono por el body
- Si no es correcto, el constraint del PhoneNumber tira el error y el mensaje.
- Si es correcto, pasaría a guardarse a la base de datos y se convierte el numero con el Converter.

Es decir, solo se utiliza este método una vez que pase del constraint.

Reply:
Entiendo, si **no hay ninguna forma de que pase el constraint y aun así no se pudiese convertir** entonces podrías dejar esa excepción aunque desconozco cuál es la mejor práctica en estos casos

Reply:
Create una custom exception como la siguiente clase, fijate que hay un package que es exceptions, podes crear un paquete dentro de ese que sea custom_exceptions
```
public class ParsePhoneNumberException extends RuntimeException {
    public ParsePhoneNumberException(String message, Throwable cause) {
        super(message, cause);
    }
}
```

y en el metodo harias lo siguiente

```suggestion
        } catch (NumberParseException e) {
            throw new ParsePhoneNumberException(e.getMessage(), e.getCause());
        }
```

Reply:
No funciona. 
Para replicar esto, tuve que sacar el constraint del numero de teléfono, y me tira un error de transacción de JPA.

Reply:
Mmmm, bueno... Me parece que no vale la pena perder el tiempo en esto por simplemente dejarlo más bonito, podemos dejarlo como deuda técnica y después vemos cómo lo modificamos